File: validation-service/src/main/java/edu/ucmo/nacha/service/ValidationServiceApplication.java
package edu.ucmo.nacha.service;

import io.dropwizard.Application;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;

public class ValidationServiceApplication extends Application<ValidationServiceConfiguration> {

    public static void main(final String[] args) throws Exception {
        new ValidationServiceApplication().run(args);
    }

    @Override
    public String getName() {
        return "ValidationService";
    }

    @Override
    public void initialize(final Bootstrap<ValidationServiceConfiguration> bootstrap) {
        // TODO: application initialization
    }

    @Override
    public void run(final ValidationServiceConfiguration configuration,
                    final Environment environment) {
        // TODO: implement application
    }

}

File: validation-service/src/main/java/edu/ucmo/nacha/service/ValidationServiceApplication.kt
package edu.ucmo.nacha.service

import com.fasterxml.jackson.module.kotlin.KotlinModule
import com.google.inject.Guice
import com.google.inject.Inject
import com.google.inject.Singleton
import edu.ucmo.nacha.file.FileModule
import edu.ucmo.nacha.record.RecordModule
import edu.ucmo.nacha.service.resources.Resource
import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment

/**
 * Validation service entry point.
 *
 * @author Grayson Kuhns
 */
@Singleton
class ValidationServiceApplication
    @Inject constructor(
        private val resources : Set<@JvmSuppressWildcards Resource>
    )
    : Application<ValidationServiceConfiguration>() {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            Guice
                .createInjector(
                    ValidationServiceModule(),
                    FileModule(),
                    RecordModule())
                .getInstance(ValidationServiceApplication::class.java)
                .run(*args)
        }
    }

    override fun getName(): String {
        return "NACHA Validation Service"
    }

    override fun initialize(bootstrap: Bootstrap<ValidationServiceConfiguration>) {
        bootstrap.objectMapper.registerModule(KotlinModule())
    }

    override fun run(config: ValidationServiceConfiguration, environment: Environment) {
        // Service API resources at /api path
        environment.jersey().urlPattern = "/api/*"

        // Register resources
        resources.forEach { environment.jersey().register(it) }
    }
}
